From a110bed7b28dbeae0cd87b03b37259119fbe1e21 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:14 -0400
Subject: [block] blk-mq: initialize resid_len

Message-id: <1402670238-13916-27-git-send-email-snitzer@redhat.com>
Patchwork-id: 83775
O-Subject: [RHEL7.1 PATCH 026/150] blk-mq: initialize resid_len
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 742ee69b92d9559e968f70cf0a49157866f8fbef
Author: Christoph Hellwig <hch@lst.de>
Date:   Mon Apr 14 10:30:06 2014 +0200

    blk-mq: initialize resid_len

    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Jens Axboe <axboe@fb.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 5455ed1..658428a 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -353,6 +353,8 @@ static void blk_mq_start_request(struct request *rq, bool last)
 
  trace_block_rq_issue(q, rq);
 
+ rq->resid_len = blk_rq_bytes(rq);
+
  /*
   * Just mark start time and set the started bit. Due to memory
   * ordering, we know we'll see the correct deadline as long as
-- 
1.7.1